### ISSUE TITLE
Identify a potential short list of additional design patterns

### ISSUE THREAD

**Issue by ash-teach** [nic-dgl104-winter-2024/pattern-library#26]
At minimum, we'll include implementations for the Singleton and Observer patterns. Use this issues to discuss additional design patterns and to create a short list. To contribute to this discussion please do the following:

1. Do a little bit of extra reading/research to identify interesting additional design patterns (perhaps on a site like [Refactoring Guru](https://refactoring.guru/design-patterns)).
2. Write a short summary (one or two paragraphs) describing the **motivation** (or purpose) of the design pattern. Include your references.
3. Include a target programming language that you think your suggested design pattern should be implemented in. 
4. [OPTIONAL] Include a list of additional programming languages for implementation, if you are aware of any.
5. Comment in this issue with your summary.

Feel free to also comment on other suggestions and design pattern ideas.

**Comment by john24alex** [comment]
@ash-teach, I would like to work on this issue if, the javascript issue is not possible.

**Comment by mijuanmontalvo** [comment]
Hello @ash-teach, Please I would like to work in this issue with @john24alex, is it possible?

**Comment by john24alex** [comment]
To cover the Identify a potential short list of additional design patterns #26 issue, I did this:

create a PR: https://github.com/nic-dgl104-winter-2024/pattern-library/pull/50



- I Create inside the patterns directory the readme with the index and the explanation of all the patterns and the idea is to redirect from that index to each pattern.

- Also, I create the abstract-factory directory with its readme file explaining the pattern and the code example.

- I modified the main readme adding the explanation of the design pattern and the link to the content inside the readme the patterns directory.

- I have a suggestion and that is to move the content of the Definition of Observer Pattern which is in the main readme to the directory of Observer Pattern inside the patterns directory.

**Comment by ash-teach** [comment]
Great. Merged #50, but I'll leave this open in case anyone else wants to add. I think @ting-dev-coder you have an open issue for a new pattern?

**Comment by nazrinzuwair** [comment]
Hi @ash-teach 
Can you assign this to me?

**Comment by john24alex** [comment]
Sure, thanks @ash-teach ,
@mijuanmontalvo  now the PR is in prod you can add your PR associated with this issue too

**Comment by mijuanmontalvo** [comment]
Ok thank you very much @ash-teach, I am going to do my part, my part will use @john24alex 's PR as a base

**Comment by mijuanmontalvo** [comment]
Ok thank you very much @john24alex

**Comment by ash-teach** [comment]
Adding the help-wanted label to let others know they are welcome to contribute new ideas here.

**Comment by mijuanmontalvo** [comment]
Hello @ash-teach, to contribute to issue #26 I did the following:

I create the PR #76, which consists of the following:

- In the "1. Creational Patterns" section table of the README.md file in the "patterns" folder add a link to the word Prototype, so that when users click, it will take them to the "Prototype" pattern folder
- 
- Inside the "patterns" folder I created a folder called "Prototype" (the folder indicated in the previous paragraph).
- 
- Inside the "Prototype" folder I created two files "README.md" and "Prototype.js".
- 
- In the "README.md" file I have placed a definition of the prototype pattern and an explanation of an example of this type of pattern for javascrip.
- 
- In the "Prototype.js" file add the example code of the prototype pattern for javascrip

**Comment by nazrinzuwair** [comment]
Hi @john24alex 

I've noticed that your README.md file primarily mentions the Factory Method design pattern instead of the Abstract Factory Method. This inconsistency could be confusing for readers who are trying to understand the specific design pattern you're focusing on. It might be beneficial to revise the references to ensure clarity and accuracy. Since Factory Method and Abstract Factory are distinct patterns, it's important to use the correct terminology to accurately convey the intended idea.

**Comment by nazrinzuwair** [comment]
Hi @ash-teach 

I've created PR #77 , which includes updates for both the Factory Design Pattern and the Facade Design Pattern:

1. Created folders named "Factory-Design-Pattern" and "Facade-Pattern" inside the PATTERN-LIBRARY directory. Each folder contains a README.md file and JavaScript files (factory-method.js and facade-pattern.js).
2. Added a README.md file in the "Factory-Design-Pattern" folder with a summary of implementing the Factory Design Pattern in JavaScript using the factory design method. Included a code snippet example related to "Creating Foods" and listed additional programming languages where the Factory Design Pattern can be implemented.
3. Included a README.md file in the "Facade-Pattern" folder with explanations and code examples for implementing the Facade Design Pattern in JavaScript.
4. Added a reference section in the README.md files of both folders with links to websites browsed during research.
5. Linked the "Factory-Design-Pattern" and "Facade-Pattern" folders into the README.md main file for easy navigation.
